lib/THC/THCGeneral.c

```diff
@@ -136,24 +142,38 @@ void THCudaEnablePeerToPeerAccess(THCState* state)
         if (can) {
           cudaError_t err = cudaDeviceEnablePeerAccess(j, 0);
           if (err == cudaErrorPeerAccessAlreadyEnabled) {
-            // Any future call to cudaGetLastError will now return an error,
-            // even though we've already dealt with this specific error here.
-            // Call cudaGetLastError once to reset the last error state.
+            /* Any future call to cudaGetLastError will now return an error, */
+            /* even though we've already dealt with this specific error here. */
+            /* Call cudaGetLastError once to reset the last error state. */
             cudaGetLastError();
 
             continue;
           }
 
           THCudaCheck(err);
+        } else {
+          /* Instead of verifying specific pairs of GPUs, we just opt out */
+          /* of p2p access. */
+          state->peerToPeerAccessEnabled = 0;
+          break;
         }
       }
     }
+
+    if (!state->peerToPeerAccessEnabled) {
+      break;
+    }
```

Comment:
I don't like this change. What if you have clusters of GPUs within a machine where GPUs have peer access within a cluster but not across clusters? You still want to use p2p within a cluster. If you're worried about cudaDeviceCanAccessPeer being slow, why don't we create a look-up table ourselves?